src/cache.jl

```diff
@@ -94,7 +94,11 @@ end
 updatecache(absentmodule::Symbol, server) = updatecache([absentmodule], server)
 
 function updatecache(absentmodules::Vector{Symbol}, server)
+    server.cp_lock && return
```

Comment:
What I had in mind was something like this:
````julia
ch = Channel()
@schedule begin
    for i in ch
        if !(missing_mod in keys(server.cache))
            updatecache(missing_mod)
        end
    end
end
````
This code would start when the language server starts, and then just keeps running as a background thread. I think it will "wake up" whenever there is something in ``ch`` to be processed. Also, when ``updatecache`` calls ``readandwrite``, i.e. starts the child process, this background thread should "fall asleep" again, and other tasks can continue processing stuff.

And then in the normal code, if one wants to add a given module to the cache, one just does
````julia
put!(ch, modulename)
````
or something like that.